app/models/dish.rb

```diff
@@ -1,4 +1,22 @@
 class Dish < ApplicationRecord
   validates_presence_of :name, :description
+
   belongs_to :chef
+  has_many :dish_ingredients
+  has_many :ingredients, through: :dish_ingredients
+
+  def total_dish_calories
+      Dish.find_by_sql("SELECT sum(dish_ingredients.quantity * ingredients.calories)
+      FROM dishes
+      JOIN dish_ingredients ON dishes.id = dish_ingredients.dish_id
+      JOIN ingredients ON ingredients.id = dish_ingredients.ingredient_id
+      WHERE dishes.id = #{self.id}").first.sum
+
+    # Dish.joins(dish_ingredients: :ingredient)
```

Comment:
Nice SQL!! You didn't need to add quantity, so what we were expecting here was just `self.ingredients.sum(:calories)` but this is awesome practice!

Your AR looks okay to me. In order to land with an integer, maybe you would just need to do `.limit(1).first.sum` in order to unwrap it. If you return to work on this and that's still not right, let me know and I can play around